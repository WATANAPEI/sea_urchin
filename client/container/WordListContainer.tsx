import React from "react";
// @ts-ignore
import useWordsApi, { WordResponse } from "../hooks/useWordsApi.tsx";
//@ts-ignore
import WordListComponent, { WordListComponentProps } from "../components/WordListComponent.tsx";

function WordListContainer({ backendUrl }: {backendUrl: string}, initialWord: WordResponse) {
  const apiPath = `${backendUrl}/words`;
  const [{ words, isLoading, isError }, doFetch] = useWordsApi(
    apiPath, [initialWord]);
  const filteredWordResponse = words.map((word: WordResponse) => {
    return {
      id: word.id,
      wordFront: word.word,
      wordBack: word.meaning
    };
  });
  const wordListComponentProps: WordListComponentProps = {
    wordListProps: filteredWordResponse,
    isLoading: isLoading,
    isError: isError
  };
  return <WordListComponent {...wordListComponentProps} />;
}

export default WordListContainer;
