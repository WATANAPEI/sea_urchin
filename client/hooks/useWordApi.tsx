import React, { useState, useEffect } from "react";
//@ts-ignore
import { translateToWordResponse } from "./wordTranslator.ts";

export interface WordResponse {
  id: number;
  word: string;
  meaning: string;
  wordLanguageID: number;
  meaningLanguageID: number;
}

export interface ReturnData {
  word: WordResponse;
  isLoading: boolean;
  isError: boolean;
}

interface UseWordApi {
  (initialUrl: string, initialWord: WordResponse): [
    ReturnData,
    React.Dispatch<React.SetStateAction<string>>
  ];
}

const useWordApi: UseWordApi = (initialUrl, initialWord) => {
  const [word, setWord] = useState(initialWord);
  const [url, setUrl] = useState(initialUrl);
  const [isLoading, setIsLoading] = useState(false);
  const [isError, setIsError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsError(false);
      setIsLoading(true);
      try {
        const response = await fetch(url);
        console.log(`reponse: ${JSON.stringify(response)}`);
        const word: WordResponse = await translateToWordResponse(response);
        setWord(word);
      } catch (error) {
        setIsError(true);
        console.log(`error: ${JSON.stringify(error)}`);
      }
      setIsLoading(false);
    };

    fetchData();
  }, [url]);

  return [{ word, isLoading, isError }, setUrl];
};

export default useWordApi;
