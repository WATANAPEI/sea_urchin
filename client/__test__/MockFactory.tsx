// @ts-ignore
import MockFetch from "./useWordApi.spec.tsx";
// @ts-ignore
import { WordResponse } from "../hooks/useWordApi.tsx";
// @ts-ignore
import { RawWordData, RawWordWithLangData } from "../hooks/wordTranslator.ts";
// @ts-ignore
import { WordWithLangResponse } from "../hooks/useWordsWithLangsApi.tsx";

export interface MockWordFactory {
  (condition: "success" | "failed", id: number, correctUrl: string): [
    WordResponse,
    MockFetch
  ];
}

export interface MockWordsFactory {
  (condition: "success" | "failed", id: number, correctUrl: string): [
    WordResponse[],
    MockFetch
  ];
}

export interface MockWordsWithLangsFactory {
  (condition: "success" | "failed", id: number, correctUrl: string): [
    RawWordWithLangData[],
    MockFetch
  ];
}

export interface MockFetch {
  (accessedUrl: string): Promise<Response>;
}

export const mockWordFactory: MockWordFactory = (condition, id, correctUrl) => {
  let mockWordResponse: WordResponse[];
  let mockResponse: Response;
  let mockFailedResponse: Promise<{}>;
  let mockSuccessFetch: MockFetch;
  let mockFailedFetch: MockFetch;
  const mockNotFoundResponse: WordResponse[] = [
    {
      id: -2,
      word: "Word Not Found",
      meaning: "Meaning Not Found",
      word_lang_id: -2,
      meaning_lang_id: -2
    }
  ];

  console.log(`mock ${id} is making`);
  if (condition === "success") {
    mockWordResponse = [
      {
        id: `${100 + id}`,
        word: `sucess word ${id}`,
        meaning: `success meaning ${id}`,
        word_lang_id: `${10 + id}`,
        meaning_lang_id: `${20 + id}`
      }
    ];
    console.log(`mock ${id} has mockWordResponse ${mockWordResponse[0].id}`);
    mockSuccessFetch = async (accessedUrl: string) => {
      if (accessedUrl == correctUrl) {
        mockResponse = new Response();
        mockResponse.json = async () => {
          console.log(`mockResponse succeed`);
          return await Promise.resolve({
            data: mockWordResponse
          });
        };
      } else {
        mockResponse = new Response();
        mockResponse.json = async () => {
          console.log(`mockResponse failed`);
          return await Promise.reject({
            data: mockNotFoundResponse
          });
        };
      }
      console.log(`mockSuccessFetch was called`);
      return await Promise.resolve(mockResponse);
    };
    return [mockWordResponse, mockSuccessFetch];
  } else {
    mockWordResponse = [
      {
        id: `${-100 - id}`,
        word: `failed word ${-1 * id}`,
        meaning: `failed meaning ${-1 * id}`,
        word_lang_id: `${-10 - id}`,
        meaning_lang_id: `${-20 - id}`
      }
    ];
    mockFailedFetch = async (accessedUrl: string) => {
      if (accessedUrl == correctUrl) {
        mockResponse = new Response();
        mockResponse.json = async () => {
          return await Promise.resolve({
            data: mockWordResponse
          });
        };
      } else {
        mockResponse = new Response();
        mockResponse.json = async () => {
          return await Promise.reject({
            data: mockNotFoundResponse
          });
        };
      }
      return await Promise.reject(mockResponse);
    };
    return [mockWordResponse, mockFailedFetch];
  }
};

export const mockWordsFactory: MockWordsFactory = (condition, id, correctUrl) => {
  let mockWordsResponse: WordResponse[] = [];
  let mockResponse: Response;
  let mockFailedResponse: Promise<{}>;
  let mockSuccessFetch: MockFetch;
  let mockFailedFetch: MockFetch;
  const mockNotFoundResponse: WordResponse[] = [
    {
      id: -2,
      word: "Word_Not_Found",
      meaning: "Meaning_Not_Found",
      word_lang_id: -2,
      meaning_lang_id: -2
    }
  ];

  //console.log(`mock ${id} is making`);
  if (condition === "success") {
    for(let i= 1; i<=10; i++){
      mockWordsResponse.push({
        id: `${100 * i + id}`,
        word: `sucess_word_${i}_${id}`,
        meaning: `success_meaning_${i}_${id}`,
        word_lang_id: `${10 * i + id}`,
        meaning_lang_id: `${20 * i + id}`
      });
    }
    //console.log(`mock ${id} has mockWordsResponse ${mockWordsResponse.id}`);
    mockSuccessFetch = async (accessedUrl: string) => {
      if (accessedUrl == correctUrl) {
        mockResponse = new Response();
        mockResponse.json = async () => {
          //console.log(`mockResponse succeed`);
          return await Promise.resolve({
            data: mockWordsResponse
          });
        };
      } else {
        mockResponse = new Response();
        mockResponse.json = async () => {
          //console.log(`mockResponse failed`);
          return await Promise.reject({
            data: mockNotFoundResponse
          });
        };
      }
      //console.log(`mockSuccessFetch was called`);
      return await Promise.resolve(mockResponse);
    };
    return [mockWordsResponse, mockSuccessFetch];
  } else {
    for( let i= 1; i <= 10; i++){
      mockWordsResponse.push({
        id: `${-100 * i - id}`,
        word: `failed_word_${i}_${id}`,
        meaning: `failed_meaning_${i}_${id}`,
        word_lang_id: `${-10 * i - id}`,
        meaning_lang_id: `${-20 * i - id}`
      });
    }
    mockFailedFetch = async (accessedUrl: string) => {
      if (accessedUrl == correctUrl) {
        mockResponse = new Response();
        mockResponse.json = async () => {
          return await Promise.resolve({
            data: mockWordsResponse
          });
        };
      } else {
        mockResponse = new Response();
        mockResponse.json = async () => {
          return await Promise.reject({
            data: mockNotFoundResponse
          });
        };
      }
      return await Promise.reject(mockResponse);
    };
    return [mockWordsResponse, mockFailedFetch];
  }
};

export const mockWordsWithLangsFactory: MockWordsWithLangsFactory = (condition, id, correctUrl) => {
  let mockWordsWithLangsResponse: RawWordWithLangData[] = [];
  let mockResponse: Response;
  let mockFailedResponse: Promise<{}>;
  let mockSuccessFetch: MockFetch;
  let mockFailedFetch: MockFetch;
  const mockNotFoundResponse: RawWordWithLangData[] = [
    {
      id: -2,
      word: "Word_Not_Found",
      meaning: "Meaning_Not_Found",
      word_language: "Word_Lang_Not_Fount",
      meaning_language: "Meaning_Lang_Not_Fount"
    }
  ];

  console.log(`mock ${id} is making`);
  if (condition === "success") {
    for (let i = 1; i<=10; i++) {
      mockWordsWithLangsResponse.push({
        id: `${100 * i + id}`,
        word: `sucess_word_${i}_${id}`,
        meaning: `success_meaning_${i}_${id}`,
        word_language: `sucess_word_lang_${i}_${id}`,
        meaning_language: `success_meaning_lang_${i}_${id}`
      });
    }
    mockSuccessFetch = async (accessedUrl: string) => {
      if (accessedUrl == correctUrl) {
        mockResponse = new Response();
        mockResponse.json = async () => {
          console.log(`mockResponse succeed`);
          return Promise.resolve({
            data: mockWordsWithLangsResponse
          });
        };
      } else {
        mockResponse = new Response();
        mockResponse.json = async () => {
          console.log(`mockResponse failed`);
          return Promise.reject({
            data: mockNotFoundResponse
          });
        };
      }
      console.log(`mockSuccessFetch was called`);
      return Promise.resolve(mockResponse);
    };
    return [mockWordsWithLangsResponse, mockSuccessFetch];
  } else {
    for (let i = 1; i <= 10; i++) {
      mockWordsWithLangsResponse.push({
        id: `${-100 * i - id}`,
        word: `failed_word_${i}_${id}`,
        meaning: `failed_meaning_${i}_${id}`,
        word_language: `failed_word_lang_${i}_${id}`,
        meaning_language: `failed_meaning_lang_${i}_${id}`
      });
    }
    mockFailedFetch = async (accessedUrl: string) => {
      if (accessedUrl == correctUrl) {
        mockResponse = new Response();
        mockResponse.json = async () => {
          return Promise.resolve({
            data: mockWordsWithLangsResponse
          });
        };
      } else {
        mockResponse = new Response();
        mockResponse.json = async () => {
          return Promise.reject({
            data: mockNotFoundResponse
          });
        };
      }
      return Promise.reject(mockResponse);
    };
    return [mockWordsWithLangsResponse, mockFailedFetch];
  }
};
