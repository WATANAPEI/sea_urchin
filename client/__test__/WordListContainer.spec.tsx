import React from "react";
import { act } from "react-dom/test-utils";
// @ts-ignore
import WordListContainer from "../container/WordListContainer.tsx";
import { createMount } from "@material-ui/core/test-utils";
// @ts-ignore
import { WordResponse } from "../hooks/useWordApi.tsx";
// @ts-ignore
import { mockWordsFactory, MockFetch } from "./MockFactory.tsx";
import "isomorphic-fetch";
import Paper from "@material-ui/core/Paper";
import { unmountComponentAtNode, render } from "react-dom";

describe("<WordListContainer />", () => {
  let container: Element;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
  });
  it("has Progress indication and Paper will appear after loading", async () => {
    const backendUrl = "http://127.0.0.1:3000";
    const initialWord = [{
      id: -1,
      word: "Initialize error",
      meaning: "Initialize error",
      wordLanguageID: -1,
      meaningLanguageID: -1
    }];
    const mockUrlList = [
        "http://127.0.0.1:3000/words"
    ];
    let mockWordsArray: WordResponse[][] = [];
    let mockFetchArray: MockFetch[] = [];
    for (let i = 0; i < mockUrlList.length; i++) {
      [mockWordsArray[i], mockFetchArray[i]] = mockWordsFactory(
        "success",
        i,
        mockUrlList[i]
      );
    }
    window.fetch = jest.fn().
      mockImplementationOnce(mockFetchArray[0]);
    const fetchSpy = jest.spyOn(window, "fetch");
    await act(async () => {
      render(<WordListContainer backendUrl={backendUrl} initialWord={initialWord} />, container);
    });
    container && console.log(container.innerHTML);
    //let paper = container.querySelector("Paper");
    //paper && console.log(paper.innerHTML);
    expect(container.querySelector("#loadingMessage")).toBeNull();
    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelectorAll("h1.word_id")).toHaveLength(mockWordsArray[0].length);
    expect(container.querySelectorAll("h3.word_word")).toHaveLength(mockWordsArray[0].length);
    expect(container.querySelectorAll("h3.word_meaning")).toHaveLength(mockWordsArray[0].length);
    expect(fetchSpy).toHaveBeenCalledTimes(1);
    expect(fetchSpy).toHaveBeenLastCalledWith(mockUrlList[0]);
  });
  it("render error message when data loading failed", async () => {
    const backendUrl = "http://127.0.0.1:3000";
    const initialWord = [{
      id: -1,
      word: "Initialize error",
      meaning: "Initialize error",
      wordLanguageID: -1,
      meaningLanguageID: -1
    }];
    const mockUrlList = [
        "http://127.0.0.1:3000/words"
    ];
    let mockWordsArray: WordResponse[][] = [];
    let mockFetchArray: MockFetch[] = [];
    for (let i = 0; i < mockUrlList.length; i++) {
      [mockWordsArray[i], mockFetchArray[i]] = mockWordsFactory(
        "failed",
        i,
        mockUrlList[i]
      );
    }
    window.fetch = jest.fn().
      mockImplementationOnce(mockFetchArray[0]);
    const fetchSpy = jest.spyOn(window, "fetch");
    await act(async () => {
      render(<WordListContainer backendUrl={backendUrl} initialWord={initialWord} />, container);
    });
    //console.log(container.innerHTML);
    expect(container.querySelector("#loadingErrorMessage")).not.toBeNull();
    expect(container.innerHTML).toMatchSnapshot();
  });
});
