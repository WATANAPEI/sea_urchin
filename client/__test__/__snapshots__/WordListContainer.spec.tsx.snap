// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordListContainer /> has Progress indication and Paper will appear after loading 1`] = `"<div class=\\"makeStyles-wordListContainer-2\\"><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"100\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">100</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_1_0</h3><h3 class=\\"word_meaning\\">success_meaning_1_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"200\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">200</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_2_0</h3><h3 class=\\"word_meaning\\">success_meaning_2_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"300\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">300</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_3_0</h3><h3 class=\\"word_meaning\\">success_meaning_3_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"400\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">400</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_4_0</h3><h3 class=\\"word_meaning\\">success_meaning_4_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"500\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">500</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_5_0</h3><h3 class=\\"word_meaning\\">success_meaning_5_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"600\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">600</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_6_0</h3><h3 class=\\"word_meaning\\">success_meaning_6_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"700\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">700</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_7_0</h3><h3 class=\\"word_meaning\\">success_meaning_7_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"800\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">800</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_8_0</h3><h3 class=\\"word_meaning\\">success_meaning_8_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"900\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">900</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_9_0</h3><h3 class=\\"word_meaning\\">success_meaning_9_0</h3></div></div></div></div></div><div class=\\"MuiPaper-root MuiPaper-elevation1 makeStyles-paper-3 MuiPaper-rounded\\" id=\\"1000\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h1 class=\\"word_id\\">1000</h1></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true\\"><h3 class=\\"word_word\\">sucess_word_10_0</h3><h3 class=\\"word_meaning\\">success_meaning_10_0</h3></div></div></div></div></div></div>"`;

exports[`<WordListContainer /> render error message when data loading failed 1`] = `"<div class=\\"makeStyles-wordListContainer-285\\"><div class=\\"MuiContainer-root makeStyles-loadingMessage-287 MuiContainer-maxWidthLg\\" id=\\"loadingErrorMessage\\">Something went wrong...</div></div>"`;
